### ISSUE TITLE
'active' flag not set for POC enrollments

### ISSUE THREAD

**Issue by cewing** [jazkarta/edx-platform#17]
When a person is enrolled in a POC, the `PocMembership` object is created with the `active` flag set to false.  This prevents the student from accessing resources they should be able to access.

This issue has blocked testing of a number of stories in github.  See the comments on story 2 in [iteration 2 from ScrumDo](https://www.scrumdo.com/projects/project/poc7/iteration/115137#) for a bit of background

**Comment by cewing** [comment]
tl;dr: 

I think I have the full scenario for this problem scouted out, except for a question about how to handle creating `PocMembership` objects when a non-user is enrolled in a POC and then registers for the site. See the very end of the following for the specifics of that question.

There are two possible scenarios here:

1.  The person being enrolled in the POC is already a registered user so there is a Django `User` object corresponding to their email or username.
2.  The person being enrolled in the POC is not a registered user so there is no Django `User` object with their email or username

The standing request is that this should work "like the instructor dashboard".  

I believe that the corresponding api action in the Instructor dashboard is [`students_update_enrollment`](https://github.com/jazkarta/edx-platform/blob/feature-pocs/lms/djangoapps/instructor/views/api.py#L414), which makes use of [`enroll_email`](https://github.com/jazkarta/edx-platform/blob/feature-pocs/lms/djangoapps/instructor/enrollment.py#L74) and [`unenroll_email`](https://github.com/jazkarta/edx-platform/blob/feature-pocs/lms/djangoapps/instructor/enrollment.py#L111) functions that are substantially similar to those that @cguardia wrote for our `pocs` django app.

In the `enroll_email` function linked above, a user who falls into category 1 above will have a new [`CourseEnrollment`](https://github.com/jazkarta/edx-platform/blob/feature-pocs/common/djangoapps/student/models.py#L685) instance created for them.  This object has an `active` flag which defaults to `True`, so this enrollment is automatically considered active when created via the Instructor Dashboard.

**So, to replicate instructor dashboard functionality for the first case, we should ensure that PocMembership objects created have the `active` flag set to `True`.**

In `enroll_email` above, a user who falls into category 2 (no Django user exists) will have a new [`CourseEnrollmentAllowed`](https://github.com/jazkarta/edx-platform/blob/feature-pocs/common/djangoapps/student/models.py#L1184) instance created for them.  This object has an `auto_enroll` boolean which defaults to `False`.  In the function, the `auto_enroll` flag on this new instance is set to the value that comes in to the api view from the dashboard template `auto enroll` checkbox.  

If email is sent to the newly enrolled user, then the email contains a link to a view that allows the student to register (which creates a new Django `User` instance).  During registration, when the user clicks on the link sent in email to 'activate` their account, the [`activate_account`]() view is hit.  This view looks up existing `CourseEnrollmentAllowed` objects and converts them to `CourseEnrollment` objects, if the `auto_enroll` flag is set to true.

So, It would appear that to replicate dashboard functionality for the second case we need to do the following:

1. Add an `auto_enroll` flag to the [`PocFutureMembership`](https://github.com/jazkarta/edx-platform/blob/feature-pocs/lms/djangoapps/pocs/models.py#L25) class
2. Connect that flag to the value for `auto enroll` passed from the poc coach dashboard to the [`poc_invite`](https://github.com/jazkarta/edx-platform/blob/feature-pocs/lms/djangoapps/pocs/views.py#L320) view.  (corresponding to the left side of the poc coach dashboard enrollment display)
3. Automatically set that flag to true for one-at-a-time additions made via the [`poc_student_management`](https://github.com/jazkarta/edx-platform/blob/feature-pocs/lms/djangoapps/pocs/views.py#L352) view (corresponding to the right side of the poc coach dashboard enrollment display)
4. Ensure that enrollments are converted at registration time, two options:
  a. Create an alternate registration flow that handles POC enrollment
  b. Extend the existing registration flow with a conditional that handles `PocFutureMembership` conversion in the existing `activate_account` view.

I believe everything is understood at this point, except for the answer to this final question.  Can I get an opinion from @pdpinch or other MIT staff regarding this question?

**Comment by cewing** [comment]
To clarify, the final question is this:

Should we enhance the existing registration flow with a conditional that activates existing `PocMembership` objects when a new user activates their account **or** should we create a new registration flow that handles it entirely separately.

**Comment by pdpinch** [comment]
I think we should add the conditional. We'll introduce less new code and our code will be closer to the core. I think it's easier to implement now & will be more robust down the line.

-Peter

----------------
I am mobile.

**Comment by pdpinch** [comment]
Oh, and everything in case 1 sounds perfect. 

----------------
I am mobile.

**Comment by cewing** [comment]
I believe this is ready to be deployed and tested now.  

You can "enroll" an existing user in a POC and they will gain access to it immediately.

You can "enroll" a non-existent user in a POC and when they register, after activating they will have access to the POC.  

The UI around showing them whether they are viewing the POC or the MOOC is still not good.  That needs work, but now the enrollment scenarios are working as expected.

Still needs tests added for views and utilities that have changed to accommodate this approach

**Comment by pdpinch** [comment]
This is functioning as expected now. I don't know the status of tests "for views and utilities that have changed to accommodate this approach" but I'm going to close this issue.

**Comment by cewing** [comment]
thanks, tests are written.  sorry not to have tagged this issue when committing them. okay to close this.
